include/graaflib/algorithm/graph_isomorphism.h

```diff
@@ -0,0 +1,482 @@
+ * @param Mapping lhs_node_order, maps vertex_id_t id of node in graph (G1) to an vertex_id_t to be easily indexed by a vector
+ * @param Mapping rhs_node_order, maps vertex_id_t id of node in graph (G2) to an vertex_id_t to be easily indexed by a vector
+ * @param Mapping lhs_reverse_node_order, reverses the mapping done by lhs_node_order for graph (G1)
+ * @param Mapping rhs_reverse_node_order, reverses the mapping done by lhs_node_order for graph (G2)
+ * @param Core_set lhs_core, a vector to map nodes from graph (G1) to graph (G2) 
+ * @param Core_set rhs_core, a vector to map nodes from graph (G2) to graph (G1)
+ * @param size_t lhs_core_len, tracks number of valid nodes in lhs_core
+ * @param size_t rhs_core_len, tracks number of valid nodes in rhs_core
+ * @param size_t tin_1_len, tracks number of valid nodes in tin_1
+ * @param size_t tout_1_len, tracks number of valid nodes in tout_1
+ * @param size_t tin_2_len, tracks number of valid nodes in tin_2
+ * @param size_t tout_2_len, tracks number of valid nodes in tout_2
+ */
+struct vf2_state {
+	Terminal_set tin_1;
+	Terminal_set tout_1;
+	Terminal_set tin_2;
+	Terminal_set tout_2;
+	Mapping lhs_node_order;  
+	Mapping rhs_node_order;  
+	Mapping lhs_reverse_node_order;  
+	Mapping rhs_reverse_node_order;  
+	Core_set lhs_core;
+	Core_set rhs_core;
+	size_t lhs_core_len = 0;
+	size_t rhs_core_len = 0;
+	size_t tin_1_len = 0;
+	size_t tout_1_len = 0;
+	size_t tin_2_len = 0;
+	size_t tout_2_len = 0;
+};	
+
+
+/**
+ * @brief a recursive helper function to facilitate creating isomorphic mapping between graphs
+ *
+ * The function will generate possible node pairs, consisting of a node from graphs lhs and rhs,
+ * and then check if a pair passes vf2's feasability rules 
+ *
+ * @tparam V The vertex type of the graphs.
+ * @tparam E The edge type of the graphs.
+ * @tparam T The graph type of the graphs (directed or undirected).
+ * @param lhs The first graph to compare.
+ * @param rhs The second graph to compare.
+ * @param state the vf2_state containing information about the mappings, core and terminal sets
+ * @param depth the depth of the recursion, important for backtracking the state to previous versions
+ * @return a bool indicating if an isomorphic mapping had been found
+ */
+template <typename V, typename E, graph_type T>
+bool check_isomorphism_helper(
+	const graph <V, E, T> & lhs,
+	const graph <V, E, T> & rhs,
+	struct vf2_state & state,
+	size_t depth
+);
+
+/**
+ * @brief initalize the inital state for the vf2 algorithm based on the graphs being compared
+ *
+ * @tparam V The vertex type of the graphs.
+ * @tparam E The edge type of the graphs.
+ * @tparam T The graph type of the graphs (directed or undirected).
+ * @param lhs The first graph to compare.
+ * @param rhs The second graph to compare.
+ * @return vf2_state struct
+ */
+template <typename V, typename E, graph_type T>
+vf2_state * init_vf2_state(
+	const graph <V, E, T> & lhs,
```

Comment:
```suggestion
	const graph <V, E, T>& lhs,
```
The convention in this project is dropping the whitespace in these cases

Reply:
ok gotcha